packages/contracts/contracts/L1/Colosseum.sol

```diff
@@ -209,108 +280,150 @@ contract Colosseum is Initializable, Semver {
+        uint256 _outputIndex,
+        bytes32 _outputRoot,
         uint256 _pos,
         Types.OutputRootProof calldata _srcOutputRootProof,
         Types.OutputRootProof calldata _dstOutputRootProof,
         Types.PublicInput calldata _publicInput,
         Types.BlockHeaderRLP calldata _rlps,
         uint256[] calldata _proof,
         uint256[] calldata _pair
-    ) external payable {
-        uint256 challengeId = latestChallengeId;
-        Types.Challenge storage challenge = challenges[challengeId];
-        _validateTurn(challenge, ValidateTurnMode.PROOF);
+    ) external {
+        Types.Challenge storage challenge = challenges[_outputIndex];
+        _validateTurn(challenge);
 
         bytes32 srcOutputRoot = Hashing.hashOutputRootProof(_srcOutputRootProof);
-        require(
-            challenge.segments[_pos] == srcOutputRoot,
-            "Colosseum: the source segment must be matched"
-        );
-
         bytes32 dstOutputRoot = Hashing.hashOutputRootProof(_dstOutputRootProof);
-        require(
-            challenge.segments[_pos + 1] != dstOutputRoot,
-            "Colosseum: the destination segment must not be matched"
-        );
+
+        // If asserter timeout, the bisection of segments may not have ended.
+        // Therefore, segment validation only proceeds when bisection is not possible.
+        if (!_isAbleToBisect(challenge)) {
+            require(
+                challenge.segments[_pos] == srcOutputRoot,
+                "Colosseum: the source segment must be matched"
+            );
+            require(
+                challenge.segments[_pos + 1] != dstOutputRoot,
+                "Colosseum: the destination segment must not be matched"
+            );
+        }
 
         require(
             _srcOutputRootProof.nextBlockHash == _dstOutputRootProof.blockHash,
             "Colosseum: the block hash must be matched"
         );
 
-        bytes32 blockHash = Hashing.hashBlockHeader(_publicInput, _rlps);
-        require(
-            _srcOutputRootProof.nextBlockHash == blockHash,
-            "Colosseum: the block hash must be matched"
-        );
+        // TODO(pangssu): waiting for the new Verifier.sol to complete.
+        // require(ZK_VERIFIER.verify(_proof, _pair), "Colosseum: invalid proof");
 
-        bytes32[] memory dummyHashes;
-        if (_publicInput.txHashes.length < MAX_TXS) {
-            dummyHashes = Hashing.generateDummyHashes(
-                DUMMY_HASH,
-                MAX_TXS - _publicInput.txHashes.length
-            );
-        }
-
-        // TODO(chokobole): check transaction root hash
-        // TODO(chokobole): use this public input hash require(_proof[4] == _publicInputHash);
-        bytes32 _publicInputHash = Hashing.hashPublicInput(
-            _srcOutputRootProof.stateRoot,
-            _publicInput,
-            CHAIN_ID,
-            dummyHashes
-        );
+        _validatePublicInput(_srcOutputRootProof, _publicInput, _rlps, _pair[4]);
```

Comment:
in `challenge.md` [line 216](https://github.com/kroma-network/kroma/pull/53/files#diff-cbfa127e2342a83c72775dbe9b975ffab0548f12513492287f8cc22e380be3b2R216), it says _proof[4] contains the public input, but here it uses _pair[4]. I wonder which one is correct.

Reply:
`_pair[4]` is correct.